slm_porosity/heat_transfer.f90

```diff
@@ -826,4 +853,28 @@ FUNCTION Spline_cubic (r_p, l_p, fr_p, fl_p, dfr_p, dfl_p)
       - fr_p*l_p**3 - r_p**2*(3*fl_p*l_p - dfl_p*l_p**2 + dfr_p*l_p**2))/(r_p-l_p)**3
   END FUNCTION Spline_cubic
 
+  FUNCTION three_parameter_model (temperature, liquid_fraction, Dsolid, Dliquid, fusion_jump)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) ::  temperature(:), liquid_fraction(:)
+    REAL (pr) :: three_parameter_model(SIZE(temperature))
+    REAL (pr) :: Dsolid, Dliquid, fusion_jump !BUT MAYBE THERE WOULD DROP INSTEAD OF JUMP HOW WE WOULD LIVE WITH THAT??//??
+    three_parameter_model = 1.0_pr + Dsolid*temperature + &
+      (fusion_jump + (Dliquid - Dsolid)*(temperature - 1.0_pr))*liquid_fraction
+  END FUNCTION three_parameter_model
+
+  FUNCTION Dthree_parameter_model (temperature, liquid_fraction, Dtemperature, Dliquid_fraction, Dsolid, Dliquid, fusion_jump)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) ::  Dtemperature(:), Dliquid_fraction(:), liquid_fraction(:), temperature(:)
```

Comment:
extra whitespace! If you are unable to cope this problem by your eyes or IDE, just try `egrep '[^, ]+  [^, ]+' heat_transfer.f90`